### PR TITLE
feat(infrastructure): generate flow types

### PR DESCRIPTION
**Pull request by gugu**

### CODE REVIEW

**Comment by gugu** [comment]
Solves #878

File: scripts/release/generate-flow.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
```
Comment (moog16) [inline scripts/release/generate-flow.js]: 2019

File: scripts/release/generate-flow.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+const isValidCwd = (
+  path.basename(process.cwd()) === 'material-components-web-react' &&
+  fs.existsSync('packages') &&
+  fs.existsSync('build')
+);
+
+if (!isValidCwd) {
+  console.error(
+    'Invalid CWD. Please ensure you are running this from the root of the repo, ' +
```
Comment (moog16) [inline scripts/release/generate-flow.js]: You should also comment `...and that you have run npm run build & scripts/release/cp-pkgs.js`

**Comment by gugu** [comment]
I found this module 5 hours ago, did not test it yet

**Comment by gugu** [comment]
No, it requires manual changes as it does not support react (and react classes in flow and ts are different)

**Comment by gugu** [comment]
I can make a different pull request with minimal set of options (same as in documentation). It will be 50 lines per module and really easy to maintain

**Comment by moog16** [comment]
@gugu  is that a different package?

**Comment by gugu** [comment]
This pull request is useless (as well as autogenerated types). I'll close it and later will create a pull request for flow-types for button. If it will be okay - I can add more

**Comment by moog16** [comment]
Yeah if you're willing to add the infrastructure in a manner that has low maintenance :) 

And as always any PR is welcome

**Pull request closed without merging**